ebpfapi/Source.def

```diff
@@ -133,3 +133,4 @@ EXPORTS
     libbpf_strerror
     ring_buffer__new
     ring_buffer__free
+    ebpf_ring_buffer_map_write
```

Comment:
Is there a libbpf equivalent of this API?

Reply:
Yes.  Why can't the existing API be used?  We should not be adding APIs specific to ebpf-for-windows when standard APIs suffice.

Reply:
The libbpf APIs are

```
LIBBPF_API struct ring_buffer *
ring_buffer__new(int map_fd, ring_buffer_sample_fn sample_cb, void *ctx,
		 const struct ring_buffer_opts *opts);
LIBBPF_API void ring_buffer__free(struct ring_buffer *rb);
LIBBPF_API int ring_buffer__add(struct ring_buffer *rb, int map_fd,
				ring_buffer_sample_fn sample_cb, void *ctx);
LIBBPF_API int ring_buffer__poll(struct ring_buffer *rb, int timeout_ms);
LIBBPF_API int ring_buffer__consume(struct ring_buffer *rb);
LIBBPF_API int ring_buffer__epoll_fd(const struct ring_buffer *rb);
```

I do not see any that lets the user mode write into the ring buffer. The APIs in the URL you share are for bpf programs to call (bpf helper functions).

I could have eBPF for Windows ringbuf map support bpf_map_update_elem instead of inventing a new API. But the problem is the length of the value is not known. Typically maps have the value_size known, but not so for ring buffer maps.

Open to suggestions.

Reply:
Did you read the reference I provided above?  E.g. bpf_ringbuf_output

Reply:
What problem is this trying to solve?  As I mentioned in a previous comment, there is no github issue linked to this.  Per ebpf-for-windows contribution guidelines, we shouldn't see a PR for major new features without an issue number we can review first.  Ring buffers were designed for exposing data from BPF programs.  Why would a user mode applicaition want to write into a ring buffer?  Why is the existing ring buffer structure the right solution for the unnamed problem?

Reply:
This is #3572.

Reply:
I responded in the issue description the other day:
"I would suggest asking on the Linux bpf list or slack channel if there is a way to do this scenario on Linux.
There might be some existing answer. Or there might be reasons to not do this but take some other approach. Before going down this road, we should understand how to make a cross-platform solution rather than designing a Windows-only one."

Issue 3572 so far does not motivate a Windows-only solution.